tests/pre-exit-report-link.bats

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env bats
+
+# To debug stubs, uncomment these lines:
+# export CURL_STUB_DEBUG=/dev/tty
+# export GIT_STUB_DEBUG=/dev/tty
+
+setup() {
+  load "$BATS_PLUGIN_PATH/load.bash"
+  export BUILDKITE_PLUGIN_TEST_COLLECTOR_API_TOKEN_ENV_NAME=""
+
+  # Config
+  export BUILDKITE_ANALYTICS_TOKEN='a-secret-analytics-token'
+  export BUILDKITE_PLUGIN_TEST_COLLECTOR_FILES='**/*/junit-1.xml'
+  export BUILDKITE_PLUGIN_TEST_COLLECTOR_FORMAT='junit'
+  # Build env
+  export BUILDKITE_BUILD_ID="an-id"
+  export BUILDKITE_BUILD_URL="https://url.com/"
+  export BUILDKITE_BRANCH="a-branch"
+  export BUILDKITE_COMMIT="a-commit"
+  export BUILDKITE_BUILD_NUMBER="123"
+  export BUILDKITE_JOB_ID="321"
+  export BUILDKITE_MESSAGE="A message"
+}
+
+COMMON_CURL_OPTIONS='--form \* --form \* --form \* --form \* --form \* --form \* --form \* --form \* --form \* --form \*'
+
+@test "Annotates report link with jq" {
+  export BUILDKITE_PLUGIN_TEST_COLLECTOR_ANNOTATE_LINK="true"
+  export CURL_RESP_FILE="./tests/fixtures/response.json"
+  stub curl "-X POST --silent --show-error --max-time 30 --form format=junit ${COMMON_CURL_OPTIONS} \* \* \* -H \* : echo 'curl success'"
```

Comment:
if you don't care about the parameters being passed, the latest versions of the mocking library allow you to do the following:

```suggestion
  stub curl "echo 'curl success'"
```